docs/discussion/lwip_ipv6.md

```diff
@@ -0,0 +1,139 @@
+# LwIP changes for CHIP
+
+LwIP is the network layer used in the CHIP platform. Although it has some good
+IPv6 support, there are areas that are lacking that we should implement for
+CHIP. The recommendations here are listed roughly from most to least important.
+
+## Route Information Options (RIO)
+
+The addressing proposal requires devices to store route options from Route
+Information Options (RIO) sent in router advertisements. This functionality is
+not currently present in upstream LwIP. The patch to add this is relatively
+small, but we may need to upstream this in order to require its use in CHIP. It
+can be added to the CHIP version, but not all platforms use the same middleware.
```

Comment:
The CHIP version is used by none of the middleware IIUC, so we don't have real tests against changes of it.